common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java

```diff
@@ -105,7 +107,11 @@ public List<ShuffleDataSegment> split(ShuffleIndexResult shuffleIndexResult) {
 
         // If ShuffleServer is flushing the file at this time, the length in the index file record may be greater
         // than the length in the actual data file, and it needs to be returned at this time to avoid EOFException
-        if (dataFileLen != -1 && totalLen >= dataFileLen) {
+        if (dataFileLen != -1 && totalLen > dataFileLen) {
+          long mask = (1L << Constants.PARTITION_ID_MAX_LENGTH) - 1;
+          LOGGER.warn("Abort inconsistent data, the data length: {}(bytes) recorded in index file is greater than "
+                  + "the real data file length: {}(bytes). Partition id: {}. This should not happen.",
```

Comment:
I feel that it will happen when the data is flushing.

Reply:
Actually I never encounter this problem with localfile storage

Reply:
> Actually I never encounter this problem with localfile storage

How about HDFS?

Reply:
Maybe... Overall this is not a breaking fix.

Reply:
Just modify the log information.

Reply:
Done